pipeline/src/pipeline/do_inference.jl

```diff
@@ -0,0 +1,19 @@
+"""
+Generate inference results using the specified truth data and pipeline.
+
+# Arguments
+- `truthdata`: The truth data used for generating inference results.
+- `pipeline`: An instance of the `AbstractAbstractEpiAwarePipeline` sub-type.
+
+# Returns
+An array of inference results.
+
+"""
+function do_inference(truthdata, pipeline::AbstractEpiAwarePipeline)
+    inference_configs = make_inference_configs(pipeline)
+    tspan = make_tspan(pipeline)
+    inference_method = make_inference_method(pipeline)
+    inference_results = map_inference_results(
```

Comment:
For the future we might want to have another module that defines paths to save results to and pass in as an arg (vs using the default) or alternative specify this via do_inference?

Reply:
or as part of the pipeline struct itself thinking about it

Reply:
Yes, I think that is a very good idea.